tmail-backend/tmail-third-party/ai-bot/src/main/java/com/linagora/tmail/mailet/rag/RagListener.java

```diff
@@ -123,10 +133,24 @@ public Publisher<Void> reactiveEvent(Event event) {
                         }
                         LOGGER.info("RAG Listener triggered for mailbox: {}", addedEvent.getMailboxId());
                         MailboxSession session = mailboxManager.createSystemSession(addedEvent.getUsername());
+                        RagondinHttpClient ragondinHttpClient = new RagondinHttpClient(ragConfig);
                         return Mono.from(messageIdManager.getMessagesReactive(addedEvent.getMessageIds(), FetchGroup.FULL_CONTENT, session))
-                            .doOnError(error -> LOGGER.error("Error occurred: ", error.getMessage()))
-                            .flatMap(this::asRagLearnableContent)
-                            .doOnNext(text -> LOGGER.info("RAG Listener successfully processed mailContent ***** \n{}\n *****", text))
+                            .doOnError(error -> LOGGER.error("Error occurred: {}", error.getMessage()))
+                            .flatMap(messageResult ->
+                                asRagLearnableContent(messageResult)
+                                    .doOnSuccess(text -> LOGGER.info("RAG Listener successfully processed mailContent ***** \n{}\n *****", text))
+                                    .flatMap(content -> new RagondinHttpClient(ragConfig)
+                                        .addDocument(
+                                            Partition.fromPattern(
+                                                ragConfig.getPartitionPattern(),
+                                                URLEncoder.encode(addedEvent.getUsername().getLocalPart(), StandardCharsets.UTF_8),
+                                                addedEvent.getUsername().getDomainPart()
+                                                    .map(Domain::asString)
+                                                    .orElseThrow(() -> new NoSuchElementException("Username must have a domain part for RAG partitioning"))),
+                                        new DocumentId(messageResult.getThreadId()),
+                                        content,
+                                        Map.of("username", addedEvent.getUsername().asString()))))
```

Comment:
We need more metadata in here 

<img width="824" height="106" alt="image" src="https://github.com/user-attachments/assets/01c6018a-3ec0-4af9-8417-d9876a2a0700" />


CF https://github.com/linagora/tmail-backend/issues/1806#issue-3170905465